var/spack/repos/builtin/packages/flang/package.py

```diff
@@ -0,0 +1,80 @@
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class Flang(CMakePackage):
+    """Flang is a Fortran compiler targeting LLVM."""
+    homepage = "https://github.com/flang-compiler/flang"
+    url      = "https://github.com/flang-compiler/flang/flecsi/tarball/v1.0"
+
+    version('develop', git='https://github.com/flang-compiler/flang', branch='master')
+
+    depends_on(
+        "llvm+clang",
+        patches=patch('https://github.com/llvm-mirror/clang/pull/33.diff',
+                      sha256='e46d7ab305e5e95c51f4656d9b52058143cd85d859b312b3c80e93a02d54b4a5',
+                      when='@4.0.1', level=1, working_dir='tools/clang'))
+
+    def patch(self):
+        # Don't use -Werror
+        # https://github.com/flang-compiler/flang/pull/85
+        filter_file(r'-Werror', '', 'CMakeLists.txt')
+
+    def cmake_args(self):
+        options = [
+            '-DCMAKE_C_COMPILER=%s' % os.path.join(
+                self.spec['llvm'].prefix.bin, 'clang'),
+            '-DCMAKE_CXX_COMPILER=%s' % os.path.join(
+                self.spec['llvm'].prefix.bin, 'clang++'),
+            '-DCMAKE_Fortran_COMPILER=%s' % os.path.join(
+                self.spec['llvm'].prefix.bin, 'flang'),
+            '-DFLANG_LIBOMP=%s' % find_libraries(
+                'libomp', root=self.spec['llvm'].prefix.lib)
+        ]
+
+        return options
+
+    @run_after('install')
+    def post_install(self):
+        # we are installing flang in a path different from llvm, so we
+        # create a wrapper with -L for e.g. libflangrti.so and -I for
+        # e.g. iso_c_binding.mod. -B is needed to help flang to find
+        # flang1 and flang2. rpath_arg is needed so that executables
+        # generated by flang can find libflang later.
+        flang = os.path.join(self.spec.prefix.bin, 'flang')
+        with open(flang, 'w') as out:
+            out.write('#!/bin/bash\n')
+            out.write(
+                '{0} -I{1} -L{2} {3}{4} -B{5} "$@"\n'.format(
+                    os.path.join(self.spec['llvm'].prefix.bin, 'flang'),
```

Comment:
So, does this modify the install directory of LLVM when it's installed?  I've been trying to build `llvm` locally to test this but I'm getting weird certificate errors at LLNL right now so I may have to track that down first.

Reply:
No, the `flang` symlink (pointing to `clang`) is build by `llvm` (with the `flang` patch), the above link just creates a more sophisticated wrapper, which incl. `-I`, `-L`, `-Wl,-rpath` and `-B`.+

Reply:
oh ok so the modifications are all handled with downstream patching.  Ok.  Still not my favorite but ok.

Just for my personal edification, how is this intended to be factored exactly?  If you call `flang` in the `llvm` install, I assume that doesn't know where its runtime libs and things are?  What is the `flang` installation providing and what is the patch providing?